src/TropicalGeometry/variety_binomial.jl

```diff
@@ -0,0 +1,42 @@
+################################################################################
+#
+#  Tropicalization of binomial ideals
+#
+#  WARNING: assumes without test that `gens(I)` is binomial
+#
+################################################################################
+
+function tropical_variety_binomial(I::MPolyIdeal,nu::TropicalSemiringMap; weighted_polyhedral_complex_only::Bool=false)
+    ###
+    # Construct matrix of exponent vector differences
+    # and vector of coefficient valuation differences
+    ###
+    G = gens(I)
+    A = matrix(ZZ,[first(collect(expv))-last(collect(expv)) for expv in exponents.(G)])
+    b = [ QQ(nu(last(collect(coeff))/first(collect(coeff)))) for coeff in coefficients.(G)]
```

Comment:
There should be no need to "collect" the exponent resp. coefficient vectors, that just introduced extra allocations. Or am I missing something?
```suggestion
    A = matrix(ZZ,[first(expv)-last(expv) for expv in exponents.(G)])
    b = [ QQ(nu(last(coeff)/first(coeff))) for coeff in coefficients.(G)]
```